report_batch/models/ir_action_report.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (C) 2019 IBM Corp.
+# Copyright (C) 2019 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+import random
+
+
+class IrActionsReportSubreport(models.Model):
+    _name = 'ir.actions.report.subreport'
+    _description = 'Report Subreport'
+    _order = 'sequence'
+
+    parent_report_id = fields.Many2one('ir.actions.report', ondelete='cascade')
+    sequence = fields.Integer(default=10)
+    model = fields.Char(related='parent_report_id.model')
+    subreport_id = fields.Many2one(
+        'ir.actions.report', string='Subreport', required=True)
+
+
+class IrActionsReport(models.Model):
+    _inherit = 'ir.actions.report'
+
+    subreport_ids = fields.One2many(
+        'ir.actions.report.subreport', 'parent_report_id')
+    model_id = fields.Many2one('ir.model', string='Model')
```

Comment:
I believe this model_id redefinition is not necessary, can you confirm?
```suggestion
```

Reply:
IMO, it may have chance where end user don't know about technical name of object. So we are providing facility to select model and it will fill up technical model name automatically.